trio/_path.py

```diff
@@ -0,0 +1,145 @@
+from functools import wraps, partial
+import os
+import types
+import pathlib
+
+import trio
+from trio._util import async_wraps
+
+
+__all__ = ['Path']
+
+
+def _forward_factory(cls, attr_name, attr):
+    @wraps(attr)
+    def wrapper(self, *args, **kwargs):
+        attr = getattr(self._wrapped, attr_name)
+        value = attr(*args, **kwargs)
+        if isinstance(value, cls._forwards):
```

Comment:
> Probably we should test some method that returns a non-PurePath? 

This is necessary, but the comparison with PurePath isn't strictly correct. I simplified this logic and added a test to trigger this case (a forwarded method that does *not* return a pathlib.Path) in  557dbdb0d6749eba6d79498b64f250c15419ff26 (`::test_forward_methods_without_rewrap`).

By pure accident, the way I wrote that test revealed a new bug:

```python
======================================== FAILURES ========================================
__________________________ test_forward_methods_without_rewrap ___________________________

path = trio.Path(/tmp/pytest-of-zack/pytest-30/test_forward_methods_without_r0/test)
tmpdir = local('/tmp/pytest-of-zack/pytest-30/test_forward_methods_without_r0')

    async def test_forward_methods_without_rewrap(path, tmpdir):
>       path = await path.parent.resolve()
E       TypeError: object PosixPath can't be used in 'await' expression

trio/tests/test_path.py:110: TypeError
========================== 1 failed, 22 passed in 0.07 seconds ===========================
```